### ISSUE TITLE
avocado raise "JobBaseException" when running "avocado run --vt-config"

### ISSUE THREAD

**Issue by PaulYuuu** [avocado-framework/avocado#3985]
There has a regression issue with https://github.com/avocado-framework/avocado/pull/3980

### master branch:
https://github.com/avocado-framework/avocado/blob/77d2f0c19c1a676a805f87578a7a13e8f60e1654/avocado/core/suite.py#L132-L144

``` shell
$ avocado run --vt-config /var/lib/avocado/data/avocado-vt/backends/qemu/cfg/tests-example.cfg --dry-run
/usr/local/lib/python3.6/site-packages/avocado_framework-80.0-py3.6.egg/avocado/plugins/run.py:285: FutureWarning: The following arguments will be changed to boolean soon: sysinfo, output-check, failfast and keep-tmp. 
  FutureWarning)

Avocado job failed: JobBaseException: Test Suite could not be create. No test references provided nor any other arguments resolved into tests
JOB HTML   : /var/tmp/avocado-dry-run-p3b3rv0v/job-2020-07-08T22.09-0000000/results.html
```

pysnooper log:
```
Source path:... /usr/local/lib/python3.6/site-packages/avocado_framework-80.0-py3.6.egg/avocado/core/suite.py
Starting var:.. cls = <class 'avocado.core.suite.TestSuite'>
Starting var:.. config = {'core.show': ['app'], 'simpletests.status.warn_...rams': None, 'vt_connect_uri': 'qemu:///session'}
Starting var:.. name = None
22:09:15.068613 call       135     def from_config(cls, config, name=None):
22:09:15.069021 line       136         references = config.get('run.references')
New var:....... references = []
22:09:15.069169 line       137         runner = config.get('run.test_runner') or 'runner'
New var:....... runner = 'runner'
22:09:15.069352 line       138         if not references:
22:09:15.069537 line       139             msg = ("Test Suite could not be create. No test references "
New var:....... msg = 'Test Suite could not be create. No test referen...ided nor any other arguments resolved into tests'
22:09:15.069689 line       141             raise TestSuiteError(msg)
22:09:15.069880 exception  141             raise TestSuiteError(msg)
Exception:..... avocado.core.suite.TestSuiteError: Test Suite co...vided nor any other arguments resolved into tests
Call ended by exception
Elapsed time: 00:00:00.001681
```

### avocado 80.0:
https://github.com/avocado-framework/avocado/blob/ac8d94ee762f4e8ff71a5cae1ddf3019f46b0595/avocado/core/job.py#L435-L457

```shell
$ avocado run --vt-config /var/lib/avocado/data/avocado-vt/backends/qemu/cfg/tests-example.cfg --dry-run
/usr/local/lib/python3.6/site-packages/avocado_framework-80.0-py3.6.egg/avocado/plugins/run.py:292: FutureWarning: The following arguments will be changed to boolean soon: sysinfo, output-check, failfast and keep-tmp. 
  FutureWarning)
JOB ID     : 0000000000000000000000000000000000000000
JOB LOG    : /var/tmp/avocado-dry-run-jwa7r3ma/job-2020-07-09T01.40-0000000/job.log
 (1/1) Host_RHEL.m8.u3.product_rhel.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.3.0.ppc64le.io-github-autotest-qemu.migrate.tcp: CANCEL: Test cancelled due to --dry-run (0.00 s)
RESULTS    : PASS 0 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 1
JOB HTML   : /var/tmp/avocado-dry-run-jwa7r3ma/job-2020-07-09T01.40-0000000/results.html
JOB TIME   : 22.89 s
```

pysnooper log:
```
Source path:... /usr/local/lib/python3.6/site-packages/avocado_framework-80.0-py3.6.egg/avocado/core/job.py
Starting var:.. self = <avocado.core.job.Job object at 0x7fff937b1c88>
01:40:54.430704 call       546     def create_test_suite(self):
01:40:54.431058 line       552         refs = self.config.get('run.references')
New var:....... refs = []
01:40:54.431131 line       553         try:
01:40:54.431240 line       554             if self._test_runner_name == 'nrunner':
01:40:54.431314 line       557                 self.test_suite = self._make_test_suite_loader(refs)
01:41:17.139511 line       562         if not self.test_suite:
01:41:17.139684 line       572         self.result.tests_total = len(self.test_suite)
01:41:17.139766 return     572         self.result.tests_total = len(self.test_suite)
Return value:.. None
Elapsed time: 00:00:22.709206
```

**Comment by beraldoleal** [comment]
@PaulYuuu thanks for reporting this, I will take a look.

**Comment by beraldoleal** [comment]
Hi @PaulYuuu, thanks for reporting this. 

Yes, we changed some internals because of the new `TestSuite()` model/class. But maybe you could help me to understand something: I couldn't find your `tests-example.cfg` file but looks like your reference is empty, right? See`refs = []`.

So how many tests do actually are running here? Because with `refs=[]` (unless I'm missing something), it seems to me that an abort it is an excepted behavior since the test suite is (in theory) empty. I also noticed that one test was canceled on 80.0. But your reference was empty, so I'm trying to understand what was your "test suite" in fact.

Since we did some internal changes I would like to understand better here before we make a decision on the course of action.

@clebergnu, probably I'm missing something, if you have any comments here too, will be more than appreciated.

**Comment by PaulYuuu** [comment]
@beraldoleal Thank you for helping so quickly.
"tests-example.cfg" includes a set of test cases and some internal configuration.

Here is its content:
```cfg
# Empty section 'top'
include base.cfg
include base-internal.cfg
include machines.cfg
vcpu_maxcpus = 64
mem = 21504
vm_create_timeout = 210
drive_cache = none
netdst = switch
setup_ksm = no
kill_vm = yes
iterations = 1
include subtests.cfg
include subtests-internal.cfg
include guest-os.cfg
include guest-hw.cfg
include guest-hw-internal.cfg
include qemu-build.cfg
include host-os-internal.cfg
include host-os-setting.cfg
include cdkeys-internal.cfg
include server-internal.cfg
# Empty section 'after_headers'

steps ?<= steps/

# Disable feature 'verify_guest_dmesg' since it is not stable yet
verify_guest_dmesg = no

only pseries
only default_bios
only RHEL.8.3.0
only ppc64le ppc64
only smallpages
only virtio_scsi
only qcow2
only bridge
only virtio_net
only vnc
only no_virtio_rng
only no_pci_assignable
only filesystem
variants:
    - @pre_test:
        only unattended_install.cdrom.extra_cdrom_ks.default_install.aio_threads rh_kernel_update
    - @run_test:
        only migrate.default.tcp.default
        # Empty section 'repeat_num'
    - @post_check:
        only check_coredump
pre_test, post_check:
    iterations = 1
no pre_test
no post_check

# Empty section 'bottom'
```

In previous versions, it looks like if "refs" is empty, the next step is to get `self.test_suite`, but now, raise the exception directly.
Not sure if there are some problems with my configuration file or use deprecated features.

Anyway, is not very urgent, we will temporarily use "80.0", thank you, and wish you a happy holiday. :smile:

**Comment by ldoktor** [comment]
Hello guys, this issue is caused by 856c3b572384eecf1d0bf1e5757cda9645843eed. Previously Avocado allowed the loader to resolve even empty suites (no positional arguments). In such case the resolver can still use arguments to decide what tests it should resolve to and that is the case of Avocado-vt's `--vt-config`. With no references the `--vt-config` is parsed and in case it resolves to any tests they are used but with the 856c3b572384eecf1d0bf1e5757cda9645843eed commit this stopped working.

A temporary workaround would be to always include at least one dummy positional argument that would be shared among all tests, for example `migrate` or `q35`.

**Comment by clebergnu** [comment]
Should be fixed by #3986 , please reopen if there are any issues left.

**Comment by beraldoleal** [comment]
Thank you @clebergnu
